File: app/src/main/java/top/easelink/lcg/ui/main/adapter/ImageAdapter.java
package top.easelink.lcg.ui.main.adapter;

import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;

import top.easelink.lcg.R;

public class ImageAdapter {

    @BindingAdapter("imageUrl")
    public static void loadImage(ImageView imageView, String url){
        Glide.with(imageView.getContext())
                .load(url)
                .placeholder(R.drawable.ic_noavatar_middle)
                .into(imageView);
    }

    @BindingAdapter("avatarUrl")
    public static void loadAvatar(ImageView imageView, String url){
        Glide.with(imageView.getContext())
                .load(url)
                .placeholder(R.drawable.ic_noavatar_middle_gray)
                .into(imageView);
    }

    @BindingAdapter("imageUrlOnly")
    public static void loadImageNoHolder(ImageView imageView, String url){
        Glide.with(imageView.getContext())
                .load(url)
                .into(imageView);
    }
}

File: app/src/main/java/top/easelink/lcg/ui/main/adapter/ImageAdapter.kt
package top.easelink.lcg.ui.main.adapter

import android.widget.ImageView
import androidx.databinding.BindingAdapter
import coil.load
import coil.transform.RoundedCornersTransformation
import top.easelink.framework.utils.dpToPx
import top.easelink.lcg.R


@BindingAdapter("avatarUrl")
fun loadAvatar(imageView: ImageView, url: String?) {
    imageView.load(url) {
        transformations(RoundedCornersTransformation(2.dpToPx(imageView.context)))
        placeholder(R.drawable.ic_noavatar_middle_gray)
    }
}